receiver/sqlserverreceiver/config.go

```diff
@@ -15,16 +16,26 @@ import (
 type Config struct {
 	scraperhelper.ControllerConfig `mapstructure:",squash"`
 	metadata.MetricsBuilderConfig  `mapstructure:",squash"`
-	InstanceName                   string `mapstructure:"instance_name"`
-	ComputerName                   string `mapstructure:"computer_name"`
+
+	InstanceName string              `mapstructure:"instance_name"`
+	ComputerName string              `mapstructure:"computer_name"`
+	Password     configopaque.String `mapstructure:"password"`
+	Port         uint                `mapstructure:"port"`
+	Server       string              `mapstructure:"server"`
+	Username     string              `mapstructure:"username"`
 }
 
 func (cfg *Config) Validate() error {
-	if cfg.InstanceName != "" && cfg.ComputerName == "" {
-		return fmt.Errorf("'instance_name' may not be specified without 'computer_name'")
+	err := cfg.validateInstanceAndComputerName()
+	if err != nil {
+		return err
 	}
-	if cfg.InstanceName == "" && cfg.ComputerName != "" {
-		return fmt.Errorf("'computer_name' may not be specified without 'instance_name'")
+
+	if !directDBConnectionEnabled(cfg) {
+		if cfg.Server != "" || cfg.Username != "" || string(cfg.Password) != "" {
```

Comment:
We don't necessarily have to return an error here, but I want to let the user know up front there may be something wrong or incomplete. There's no use-case for specifying some, but not all, of the connection options, and I could see it being a source of confusion if a isn't seeing metrics because they unknowingly forgot a configuration option.